src/build.d

```diff
@@ -414,6 +415,54 @@ alias runDmdUnittest = memoize!(function()
     return new DependencyRef(dep);
 });
 
+/// Checks all sources for whitespace errors
+alias checkWhitespace = memoize!(function()
+{
+    Dependency dep;
+    with (dep)
+    {
+        name = "checkwhitespace";
+        description = "Check the sources for whitespace errors";
+        msg = "(RUN) CHECK-WHITESPACE";
+        deps = [tools];
+        command = [
+            env["HOST_DMD_RUN"],
+            "-run",
+            env["TOOLS_DIR"] ~ "/checkwhitespace",
+            "--allow-windows-newlines"
+        ].chain(
+            sourceFiles.sources,
+            sourceFiles.glue,
+            sourceFiles.root,
+            // Disabled on Posix - why?
+            env["OS"] == "windows" ? sourceFiles.backend : []
+        ).array;
+
+        // TODO: Run with CC=HOST_CXX in posix.mak - seems irrelevant?
+    }
+    return new DependencyRef(dep);
+});
+
+/// Clones GIT_HOME/tools if TOOLS_DIR does not exist
+alias tools = memoize!(function()
+{
+    Dependency dep;
+    with (dep)
+    {
+        name = "tools";
+        description = "Clone the tools repository";
+        msg = "(GIT) Cloning tools";
+        commandFunction = ()
+        {
+            // Use explicit exists instead of target because isUpToDate considers build.d's timespamp
+            if(!exists(env["TOOLS_DIR"] ~ "/checkwhitespace.d")) {
+                [env["GIT"], "clone", "--depth=1", env["GIT_HOME"] ~ "/tools", env["TOOLS_DIR"]].run();
```

Comment:
Why not the traditional `run(...)`.

Reply:
Personal preference, but i will change it.